### ISSUE TITLE
feature/function-pathexist

### ISSUE THREAD

**Issue by kobmaki** [Icinga/icinga2#5510]
Actually there is no possibility to test inside icinga2 if a path exist. With the ITL comes a lot of checks that required installations of add ons. 

There are simliar function like basename or dirname that are usefull functions.

## Expected Behavior
pathexist("/") -> true
pathexist("/a/path/to/a/file/or/directory/not/exist")  -> false
```
<1> => pathexist("/")
true
<2> => pathexist("/a/path/to/a/file/or/directory/not/exist")
false
<3>
```

## Current Behavior
Function not exist.

```
root@deb-icinga-test:~# icinga2 console
Icinga 2 (version: v2.7.0)
<1> => pathexist("/");
       ^^^^^^^^^^^^^^
Argument is not a callable object.
<2> => 
```

## Possible Solution
Make the function PathExist available as pathexist from  lib/base/scriptutils.cpp and lib/base/utility.hpp

I already have a patch for this feature include the documentation. If the feature is accepted I will create a branch and send a pull request.

## Steps to Reproduce (for bugs)
No bug, just a feature request.

## Context
Find, which checks are usable on the machine. 
I try to find a check that is correct defined but the real check execution  from an addon is not available. The reason could be missing packages or simple an error in the definition.

The following simple code will create a dictionary with  name of the CheckCommand that has the value true if the the path exist, otherwise false::

```
x={}; 

for (a in get_objects(CheckCommand) ) { 
   x.set(a.name, pathexist(a.command[0])); 
  };
```

Anothet question is to check if some global directory exist:

```
y={}
for ( a in globals.keys().filter( e => (match("*Dir",e)) ) ) {
  y.set(a,pathexist( globals.get(a)));
}
 ```

Both example are very helpful in finding configuration errors.

## Your Environment

```
root@deb-icinga-test:~# icinga2 --version
icinga2 - The Icinga 2 network monitoring daemon (version: v2.7.0)

Copyright (c) 2012-2017 Icinga Development Team (https://www.icinga.com/)
License GPLv2+: GNU GPL version 2 or later <http://gnu.org/licenses/gpl2.html>
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.

Application information:
  Installation root: /usr
  Sysconf directory: /etc
  Run directory: /run
  Local state directory: /var
  Package data directory: /usr/share/icinga2
  State path: /var/lib/icinga2/icinga2.state
  Modified attributes path: /var/lib/icinga2/modified-attributes.conf
  Objects path: /var/cache/icinga2/icinga2.debug
  Vars path: /var/cache/icinga2/icinga2.vars
  PID path: /run/icinga2/icinga2.pid

System information:
  Platform: Debian GNU/Linux
  Platform version: 8 (jessie)
  Kernel: Linux
  Kernel version: 4.9.0-0.bpo.3-amd64
  Architecture: x86_64

Build information:
  Compiler: GNU 4.9.2
  Build host: 69a566ab7b7f
root@deb-icinga-test:~# 
```

```
egrep "NAME|VERSION|ID" /etc/os-release 

PRETTY_NAME="Debian GNU/Linux 8 (jessie)"
NAME="Debian GNU/Linux"
VERSION_ID="8"
VERSION="8 (jessie)"
ID=debian
```

**Comment by dnsmichi** [comment]
I agree on the plugins path idea, but using this function allows/invites to test for "any" file existance. This is a similar request like exec() or any other system calls which we've denied for security reasons. basename/dirname should remain an exception and not become a rule imho.

**Comment by gunnarbeutner** [comment]
There's no security issue here, really. Besides, you can sort-of (with a big caveat) build this functionality yourself:

```
try {
  include "/path"
} except {
  log("File does not exist.")
}
``` 

I'll see what I can do about this feature request. :)

**Comment by dnsmichi** [comment]
Closed in favor of #5580.
